lib/phoenix_live_view/test/dom.ex

```diff
@@ -328,6 +328,9 @@ defmodule Phoenix.LiveViewTest.DOM do
       |> Enum.map(fn {attr, value} -> {String.downcase(to_string(attr)), value} end)
       |> Enum.filter(fn {attr, _value} -> attr not in reserved_attrs end)
       |> Enum.reduce(container_attrs_list, fn {attr, new_val}, acc ->
+        new_val =
+          if attr == "class" and is_list(new_val), do: Enum.join(new_val, " "), else: new_val
```

Comment:
Thanks for the PR! I think it would be best to guarantee the server always sends it as a string, so we don't need this logic here. Do we have a similar logic in .js too?

Reply:
I was thinking the same.

As far as I can tell, and it's been a while since I did the debugging, the frontend logic is in `dom.js` `replaceRootContainer` which only blindly maps over the attributes and doesn't check value.

So yes, that would be an argument in favor of removing the change from `dom.ex`. Is that what you mean? That would mean basically reverting one of the two commits here.

Reply:
I will do a quick check and come back here

Reply:
Yes, reverting the dom.ex change. And making sure that we don't pass a list anywhere else :)

Reply:
We retrieve the container using `Route.container/1` and this function is only ever called from this one location.

```elixir
# Route.container/1

def container(%Route{} = route) do
  route.opts[:container] || route.view.__live__()[:container]
end
```

Any other reference to container is in 

- `__before_compile__` of `Phoenix.Component` where we pass it forward with a default ✅ safe

```elixir
    container = opts[:container] || {:div, []}

    live = %{
      container: container,
      name: name,
      kind: :view,
      module: env.module,
      layout: layout,
      lifecycle: lifecycle,
      log: log
    }
```

- `Router.validate_live_opts!` where we just validate opts are a list ✅ safe

```Elixir
Enum.each(opts, fn
  {:container, {tag, attrs}} when is_atom(tag) and is_list(attrs) ->
    :ok

{:container, val} ->
  raise ArgumentError, """
  expected live :container to be a tuple matching {atom, attrs :: list}, got: #{inspect(val)}
  """
```

- `Static.container` (private) where we just merge some options

This is used in `render` and `nested_render` which would I assume do the initial rendering server-side using Phoenix.HTML? I also assume that then handles a class given as a list internally, since that was, I think, my experience with classic phoenix. So i'm gonna go with ☑️ sefe-ish?

```Elixir
defp container(%{container: {tag, attrs}}, opts) do
  case opts[:container] do
    {tag, extra} -> {tag, Keyword.merge(attrs, extra)}
    nil -> {tag, attrs}
  end
end
```